sql/core.go

```diff
@@ -283,8 +285,57 @@ type RowUpdater interface {
 // Database represents the database.
 type Database interface {
 	Nameable
-	// Tables returns the information of all tables.
-	Tables() map[string]Table
+
+	// GetTableInsensitive retrieves a table by it's name where capitalization does not matter
+	GetTableInsensitive(ctx context.Context, tblName string) (Table, bool, error)
```

Comment:
Spell out how the case sensitivity should work: prefer an exact match, fall back to single case insensitive match if there isn't a case sensitive match, error if there is more than one case-insensitive match. Probably mention the GetTableInsensitive method below. Also mention the two non-obvious return params.

Reply:
updated.